op-node/p2p/peer_scorer.go

```diff
@@ -1,16 +1,92 @@
 package p2p
 
 import (
+	"fmt"
+	"strconv"
+	"strings"
+
 	log "github.com/ethereum/go-ethereum/log"
 	pubsub "github.com/libp2p/go-libp2p-pubsub"
 	peer "github.com/libp2p/go-libp2p/core/peer"
 )
 
 type scorer struct {
-	peerStore Peerstore
-	metricer  GossipMetricer
-	log       log.Logger
-	gater     PeerGater
+	peerStore           Peerstore
+	metricer            GossipMetricer
+	log                 log.Logger
+	gater               PeerGater
+	bandScoreThresholds BandScorer
+}
+
+// bandScoreThresholds holds the thresholds for classifying peers
+// into different score bands.
+type bandScoreThresholds struct {
+	bands      map[string]float64
+	lowestBand string
+}
+
+// BandScorer is an interface for placing peer scores
+// into various bands.
+//
+// Implementations are expected to construct internals using the
+// [Parse] function and then expose the [Bucket] function for
+// downstream [BandScorer] consumers.
+//
+//go:generate mockery --name BandScorer --output mocks/
+type BandScorer interface {
+	Parse(str string) error
+	Bucket(score float64) string
+	Reset()
+}
+
+// NewBandScorer constructs a new [BandScorer] instance.
+func NewBandScorer() BandScorer {
```

Comment:
It is good practice in Go to _return_ concrete implementations while _consuming_ interfaces, unless there's a specific reason to return an interface. There are many write ups on that, like [this one](https://medium.com/@cep21/preemptive-interface-anti-pattern-in-go-54c18ac0668a).

Here, it probably makes sense to make the `bandScoreThresholds` type public and return it. Then you also don't need to convert the interface to the concrete type in the tests.